hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -112,8 +115,21 @@ private String makeWriteToken() {
     return FSUtils.makeWriteToken(getPartitionId(), getStageId(), getAttemptId());
   }
 
+  protected StoragePath getPartitionPath(String partitionPath) {
+    boolean isMDT = hoodieTable.getMetaClient().getTableConfig().getTableName().contains("metadata");
+    if (isMDT) {
+      return FSUtils.constructAbsolutePath(config.getBasePath(), partitionPath);
+    } else {
+      HashMap<String, String> configMap = new HashMap<>();
+      Path basePath = new Path(config.getBasePath());
+      configMap.put(HoodieOSSStorageStrategy.FILE_ID_KEY, fileId);
+      configMap.put(HoodieOSSStorageStrategy.TABLE_BASE_PATH, basePath.toUri().getPath());
+      return storage.storageLocation(partitionPath, configMap);
+    }
+  }
+
   public StoragePath makeNewPath(String partitionPath) {
-    StoragePath path = FSUtils.constructAbsolutePath(config.getBasePath(), partitionPath);
+    StoragePath path = getPartitionPath(partitionPath);
```

Comment:
I assume this change is just for the POC? Ideally the conversion from logical path to physical path should happen within `HoodieStorage` at L134